### PR TITLE
[stable/redis-ha] replace slave deployment to statefulset + create separate pvc for masters and slaves 

### PR DESCRIPTION
**Pull request by smileisak**
Changelog:
* Statefulset instead of deployment for slave component.
* Separate pvc for each role (master, slave)

Related with these issues && PR: #2527 #2543 https://github.com/kubernetes/charts/issues/2539 https://github.com/kubernetes/charts/issues/2386

### CODE REVIEW

**Comment by smileisak** [comment]
@viglesiasce there is some conflicts that i can't resolve. `Only those with write access to this repository can merge pull requests.`

**Comment by viglesiasce** [comment]
@smileisak you need to pull the latest from master then in your branch do a `git merge origin/master`. Can you show me what you were doing?

**Comment by smileisak** [comment]
@viglesiasce i was adding only the 2 last commits but it take all from previous PR: https://github.com/kubernetes/charts/pull/2543

**Comment by smileisak** [comment]
@viglesiasce conflicts fixed :dancing_men: . Thank you :+1:.

File: stable/redis-ha/Chart.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ keywords:
 - redis
 - keyvalue
 - database
-version: 0.2.3
+version: 0.3.0
```
Comment (viglesiasce) [inline stable/redis-ha/Chart.yaml]: This is a breaking change so should bump this up to 1.x.x

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: stable/redis-ha/Chart.yaml (hunk 1 above)
Comment (smileisak) [reply stable/redis-ha/Chart.yaml]: Updated to 1.0.0 :smile_cat:

**Review by viglesiasce — APPROVED** [review]

**Pull request merged**